sympy/calculus/util.py

```diff
@@ -1631,3 +1632,178 @@ def union(self, other):
 
 # setting an alias for AccumulationBounds
 AccumBounds = AccumulationBounds
+
+def argmax_real(f, symbol = None, domain = S.Reals):
+    """
+    Returns the set of arguments of maxima, i.e., values of `symbol`
+    for which the `f` attains the maximum value in the given domain.
+    Note: This function uses derivatives to calculate the points where the
+    maximum value is achieved, and thus, would not include points where the
+    function is not differentiable. Also, the domain should be a subset of
+    S.Reals
+
+    Parameters
+    ==========
+
+    f: expr
+        The concerned expression
+    symbol: Symbol
+        Variable for which the arguments to be determined.
+    domain: Set
+        The domain over which the arguments are to be determined
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, sin, Interval, S, exp, pi
+    >>> from sympy.calculus.util import argmax_real
+
+    >>> x = Symbol('x')
+    >>> argmax_real(sin(x), x, Interval(-2*pi, pi))
+    FiniteSet(-3*pi/2, pi/2)
+
+    >>> argmax_real(-x**2 + 5*x - 6)
+    FiniteSet(5/2)
+
+    >>> argmax_real(exp(x), x, S.Reals)
+    EmptySet
+    """
+    f = _sympify(f)
+    if symbol is None:
+        if len(f.free_symbols) > 1:
+            raise ValueError(filldedent('''
```

Comment:
I guess this should be `len(free_symbols) != 1`.

Maybe this logic could be moved to `_argMaxMin` if it otherwise needs to be duplicated in `argmax_real` and `argmin_real`